mslib/msui/mscolab.py

```diff
@@ -536,11 +536,13 @@ def after_login(self, emailid, url, r):
 
     def fetch_gravatar(self, refresh=False):
         email_hash = hashlib.md5(bytes(self.email.encode('utf-8')).lower()).hexdigest()
-        email_in_settings = self.email in config_loader(dataset="gravatar_ids")
+        email_in_config = self.email in config_loader(dataset="gravatar_ids")
         gravatar_path = os.path.join(constants.MSS_CONFIG_PATH, 'gravatars')
         gravatar = os.path.join(gravatar_path, f"{email_hash}.png")
 
-        if refresh or email_in_settings:
+        # refresh is used to fetch new gravatar associated with the email
+        if refresh or email_in_config:
+            # create directory to store cached gravatar images
             if not os.path.exists(gravatar_path):
                 try:
                     os.makedirs(gravatar_path)
```

Comment:
we should use here **fs** to create and test the directory. On running tests this currently creates a directory outside of tempfs. 
On my linux system it creates various dirs similiar to  osfs:/tmp/tmp9y4tzpcemssf8b90094fc in the MSS folder because os.makedirs does not understand fs syntax.

Reply:
I tried using fs.open_fs for creating the 'gravatars' folder, but it doesn't seem to work. How else can I create a dir using fs?

Reply:
e.g.
```
home_fs = fs.open_fs("~/")
if not home_fs.exists("gravatars"):
    home_fs.makedirs("gravatars")
```
it may be is also a good idea to store the gravatars dir in the .config/mss folder?